mrtt-ui/src/components/CheckboxGroup/CheckboxGroup.stories.js

```diff
@@ -39,3 +39,14 @@ export const OtherOptionPrePopulated = () => (
     shouldAddOtherOptionWithClarification={true}
   />
 )
+
+export const WithNestedMarkupForSelected = () => (
+  <CheckboxGroup
+    options={options}
+    value={{ selectedValues: ['Option 1', 'Option 2'], otherValue: 'Some other value' }}
+    onChange={action('onChange')}
+    id='cuddlyKittens'
+    shouldAddOtherOptionWithClarification={true}
+    SelectedMarkup={() => <>Youve selected ME</>}
```

Comment:
SelectedMarkup is basically the big change to CheckboxGroup. Whatever you put here, will show for any checkbox checked (always the same markup - wont work for nested stuff that is unique for each checkbox)

Reply:
hmm what would be a big change? would this be like my section 4 deep nesting structure? what would be an example of how to use this?

Reply:
See `mrtt-ui/src/components/RestorationAimsForm/StakeholderBenefitsInputs.js` and `mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js`. You can also run storybook and take a look. This story is the most basic example. 

this is basically inversion of control so that we can control the nested children for each option from a composing parent - to keep checkbox group generic and reusable (although lets be honest, its probably reaching the max limit for complexity for a generic)

Reply:
ok cool! yes I agree that is probably reaching max complexity